feat: bridging to Sapphire (#1328)
oasisprotocol/docs: docs/general/manage-tokens/how-to-transfer-rose-erc20.md

# How to Bridge Assets to Oasis Sapphire

This guide shows how to **bridge assets (ETH, USDC, USDT, etc.)** from networks
like Ethereum, BNB Chain, or Polygon to **Oasis Sapphire** - an EVM-compatible
ParaTime with confidential smart contracts.

**Main Bridge:** [**Celer cBridge**][cbridge] - a decentralized, non-custodial
bridge supporting fast transfers from multiple networks.

This guide covers:

- Bridging assets via cBridge
- Transferring ROSE to Sapphire
- Important warnings and FAQs

**Prerequisites:**

- Web3 wallet (e.g. MetaMask) with Oasis Sapphire network added
- Gas fees on source chain (ETH, BNB, etc.)

> :::tip
> Have some **ROSE on Sapphire** for gas fees to use your bridged tokens.
> See [Transferring ROSE](#transferring-rose-to-sapphire-from-exchanges-or-oasis-wallet) below.
> :::

Reviewer: I'm not sure if I understood this admonition correctly. If a user needs native ROSE, they should skip the first section?

Anyway, I would point them to https://docs.oasis.io/general/manage-tokens/#obtaining-rose

```suggestion
:::info Need some ROSE on Sapphire for gas fees to use your bridged tokens?

See [Transferring ROSE](#transferring-rose-to-sapphire-from-exchanges-or-oasis-wallet) below.

:::
```

Author: Reading all comments, the conclusion would be:
1. point in the admonition to #obtaining-rose
2. rename the section #get-rose
3. extent #get-rose section with parts form #tranfser-rose from this page?
then on new issue:
4. move get-rose to own page?
5. revamp get-rose and integrate https://oasis.net/blog/onboarding-guide-rose-sapphire ?

> 

Reviewer: We can mention the Celer IM chapter too.

```suggestion
:::tip Are you a developer?

Celer cBridge uses Celer Inter-Chain Messaging (IM) protocol to communicate between the chains. You can learn how build cross-chain dApps on Oasis using OPL and Celer IM [here](../../build/opl/celer/README.md).

:::
```

        wrapped ROSE token on BNB Chain).

4.  **Enter Amount and Transfer:** Enter the amount you wish to bridge. The
    interface may display the estimated receive amount after fees. Click
    **Transfer**. For ERC-20 assets, your wallet will first
    ask you to **approve** the cBridge contract to spend that token (e.g.
    approve USDC). Approve the token, then confirm the **bridging transaction**
    in your wallet. This will send the tokens to cBridge on the source chain.

5.  **Confirm and Wait:** After confirming, cBridge will handle the cross-chain
    transfer. The bridging typically completes **within a few minutes**, but
    times vary by chain. You can monitor the transfer status on the cBridge
    interface.

6.  **Receive Tokens on Sapphire:** Tokens arrive at your Sapphire address.
    You may need to add the token contract address in MetaMask to see them.
    See [Contract Addresses][token-addresses] for official token addresses.

:::info
cBridge is **non-custodial** - tokens are sent automatically after
confirmation. Always use the **official URL** and verify **Oasis Sapphire** as
destination.
:::

> ## Transferring ROSE to Sapphire (From Exchanges or Oasis Wallet)

Reviewer: This is (or should) already be covered at https://deploy-preview-1328--oasisprotocol-docs.netlify.app/general/manage-tokens/#obtaining-rose. Maybe we should call the section there "Get ROSE" or maybe we could even devote a special chapter to it. In any case can you move this section there and just link it from here?

You need ROSE on Sapphire for gas fees. Get ROSE from:

### From a Centralized Exchange via ROSE App

For ROSE on exchanges (Binance, Coinbase, etc.), use the **ROSE App**:

1.  **Open the ROSE App Move interface:** Visit

>     [**rose.oasis.io/\#/move**](https://rose.oasis.io/#/move) in a web browser.

Reviewer: This has changed recently.

```suggestion
    [**rose.oasis.io/\#/move**](https://rose.oasis.io/move) in a web browser.
```

token contract address from our [Contract Addresses][token-addresses] page.

Use your tokens in Sapphire dApps like [Neby][neby] for swapping, liquidity
provision, and more.

## Warnings and Best Practices

*   **Use Official Links:** Only use official [cBridge][cbridge] URL - beware
    of phishing sites.

*   **Gas Fees:** Need native tokens on source chain + ROSE on Sapphire.
    Without ROSE on Sapphire, you cannot use your bridged tokens.

*   **Transaction Time:** Transfers take a few minutes. Check cBridge status
    if delayed beyond 10-15 minutes.

*   **Hardware Wallets:** Enable "Blind Signing" or "Contract Data" in
    Ledger's Ethereum app for bridge transactions.

*   **Test First:** Do a small test transfer before large amounts.

*   **Fees:** cBridge charges small fees. Check minimum/maximum limits
    in the interface.

*   **Security:** Always bridge to your own wallet first, not directly
    to dApp contracts.

*   **Token Versions:** Verify dApps support your bridged token version.

*   **Mistakes:** Contact bridge support immediately if sent to wrong
    network. Always verify **Oasis Sapphire (chain ID 23294)**.

> ## FAQs

Reviewer: We should probably move this section under https://deploy-preview-1328--oasisprotocol-docs.netlify.app/general/manage-tokens/faq and split it to different sections similar to https://deploy-preview-1328--oasisprotocol-docs.netlify.app/general/oasis-network/faq

<details>

> <summary>**Do I need ROSE in my wallet to bridge assets to Sapphire?**</summary>

Reviewer: No need for bold IMO. Also below.

```suggestion
<summary>Do I need ROSE in my wallet to bridge assets to Sapphire?</summary>
```

Author: I changed the structure to match the other faq style

<summary>**Are there any fees to bridge apart from network gas fees?**</summary>

Yes, small bridge fees plus network gas. Ethereum gas is usually the bigger cost
compared to Polygon or BNB Chain. Check fee info in the interface. Some popular
routes may have 0% fees.
</details>

<details>
<summary>**What is the difference between using the ROSE App vs. using a bridge?**</summary>

**ROSE App**: Only for ROSE transfers between consensus layer/exchanges and
Sapphire. Not for other tokens.

**cBridge**: For all tokens including ETH, USDC, etc. Also works for ROSE from
BNB Chain.
</details>

<details>
<summary>**The token I bridged has “.e” at the end (like USDC.e). What does that
mean?**</summary>

".e" means "bridged from Ethereum" - helps distinguish token versions. Same
underlying asset, different contracts. Check which version your dApp supports.
</details>

<details>
<summary>**Can I move my bridged tokens back to the original chain?**</summary>

Yes, use cBridge in reverse: select Sapphire as source, original chain as
destination. Need ROSE for gas on Sapphire.
</details>

<details>
<summary>**The Wormhole bridge was used in older docs – is that still available?**</summary>

> Wormhole was used for Emerald ParaTime. Sapphire uses **Celer cBridge** as the

Reviewer: ```suggestion
Wormhole was used for Emerald ParaTime and is considered deprecated (archived instructions are
available [here](https://github.com/oasisprotocol/docs/blob/01cee79bf75fbda1a2f74543f2cf24ccd25eeed1/docs/general/manage-tokens/how-to-transfer-eth-erc20-to-emerald-paratime.md)).
Sapphire uses **Celer cBridge** as the
```
